src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -41,15 +43,17 @@ public bool TryParseMessages(ReadOnlyMemory<byte> input, IInvocationBinder binde
         {
             while (BinaryMessageParser.TryParseMessage(ref input, out var payload))
             {
-                messages.Add(ParseMessage(payload.ToArray(), binder));
+                var isArray = MemoryMarshal.TryGetArray(payload, out var arraySegment);
+                Debug.Assert(isArray);
```

Comment:
Add a comment saying when TryGetArray would return false and why it will never happen here

Reply:
It'll happen the day we switch to use native buffers 😄(never)

Reply:
Done

Reply:
I like this @JamesNK fella